File: benchmarks/workloads/regex_match_groups/javascript/bench.js
const n = parseInt(process.argv[2] || "3000000", 10);

// Named capture groups, anchored full-match.
const re = /^(?<l>[0-9]+)x(?<w>[0-9]+)x(?<h>[0-9]+)$/;
const inputs = ["2x3x4", "20x3x11", "hello world!"];

let matched = 0, none = 0, total = 0;
for (let i = 0; i < n; i++) {
    const s = inputs[i % 3];
    const m = re.exec(s);
    if (m) {
        const l = +m.groups.l, w = +m.groups.w, h = +m.groups.h;
        total += l * w * h;
        matched++;
    } else {
        none++;
    }
}
console.log(`matched = ${matched} none = ${none} total = ${total}`);

File: benchmarks/workloads/regex_match_groups/go/bench.go
package main

import (
	"fmt"
	"os"
	"regexp"
	"strconv"
)

func main() {
	n := 3000000
	if len(os.Args) > 1 {
		if v, err := strconv.Atoi(os.Args[1]); err == nil {
			n = v
		}
	}

	// Named capture groups (Go spells them (?P<name>...)), anchored full-match.
	re := regexp.MustCompile(`^(?P<l>[0-9]+)x(?P<w>[0-9]+)x(?P<h>[0-9]+)$`)
	inputs := []string{"2x3x4", "20x3x11", "hello world!"}

	var matched, none, total uint64
	for i := 0; i < n; i++ {
		s := inputs[i%3]
		m := re.FindStringSubmatch(s)
		if m != nil {
			l, _ := strconv.ParseUint(m[1], 10, 64)
			w, _ := strconv.ParseUint(m[2], 10, 64)
			h, _ := strconv.ParseUint(m[3], 10, 64)
			total += l * w * h
			matched++
		} else {
			none++
		}
	}
	fmt.Printf("matched = %d none = %d total = %d\n", matched, none, total)
}
